iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmPreprocessor.java

```diff
@@ -38,14 +38,18 @@ public static String addLineComments(String content) {
     var lineCounter = 0;
     var matcher = LINE_PATTERN.matcher(content);
 
+    int lastIndex = 0;
     while (matcher.find()) {
       lineCounter++;
       String lineContent = matcher.group("lineContent");
       String comment = commentLineNumber(lineCounter);
       String newLine = matcher.group("newLine");
-      matcher.appendReplacement(sb, lineContent + comment + newLine);
+      sb.append(lineContent);
+      sb.append(comment);
+      sb.append(newLine);
```

Comment:
I think we don't need to save them as variables and can directly write `sb.append(matcher.group("newline"))`